src/api/app/services/monitor_controller_service/status_history_fetcher.rb

```diff
@@ -0,0 +1,32 @@
+module MonitorControllerService
+  class StatusHistoryFetcher
+    def initialize(key, range)
+      @key = key
+      @range = user_range(range)
+    end
+
+    def call
+      Rails.cache.fetch(cache_key, expires_in: (@range.to_i * 3600) / 150, shared: true) do
+        status_history
+      end
+    end
+
+    private
+
+    def user_range(range)
+      [upper_range_limit, range].min
+    end
+
+    def upper_range_limit
+      24 * 365
```

Comment:
Same magic numbers here. This time I have no idea what this would mean. 24 years? I also don't understand why we support arbitrary ranges if the interface only ever asks for 

![Screenshot from 2020-01-29 12-52-39](https://user-images.githubusercontent.com/514785/73354544-421dd380-4296-11ea-8a6b-869100df7ea2.png)

Reply:
I totally understand if you don't want to go into this so deep...

Reply:
I would love definitely to work on that but not in this PR.  I think that was just an arbitrary value chosen to make sure that `[upper_range_limit, user_range].min` works as expected.